src/app/clusters/service-area-server/service-area-delegate.h

```diff
@@ -85,7 +75,9 @@ class Delegate
      *
      * @param[in] selectedAreas List of new selected locations.
      * @param[out] locationStatus Success if all checks pass, error code if failure.
-     * @param[out] statusText text describing failure (see description above), size kMaxSizeStatusText.
+     * @param[out] statusText text describing failure (see description above). Max size kMaxSizeStatusText.
+     * Note: If the return is false and statusText is not successfully edited, example due to a string that can be longer than
+     * kMaxSizeStatusText, the size of this value should be set to 0 with .reduce_size(0) to avoid memory leaks.
```

Comment:
```suggestion
     * Note: If the return is false and statusText is not successfully set, for example due to a string that can be longer than
     * kMaxSizeStatusText, the size of this value should be set to 0 with .reduce_size(0) to avoid callers using un-initialized memory.
```